rerun_py/rerun_sdk/rerun/archetypes/image_ext.py

```diff
@@ -1,19 +1,80 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from io import BytesIO
+from typing import TYPE_CHECKING, cast
 
 import numpy as np
 import pyarrow as pa
 
 from .._validators import find_non_empty_dim_indices
+from ..datatypes import TensorBufferType
 from ..error_utils import _send_warning, catch_and_log_exceptions
 
 if TYPE_CHECKING:
+    from .._image import ImageEncoded
     from ..components import TensorDataBatch
     from ..datatypes import TensorDataArrayLike
+    from . import Image
 
 
 class ImageExt:
+    JPEG_TYPE_ID = list(f.name for f in TensorBufferType().storage_type).index("JPEG")
+
+    def compress(self, *, jpeg_quality: int = 95) -> ImageEncoded | Image:
+        """
+        Converts an `Image` to an `ImageEncoded` using JPEG compression.
+
+        JPEG compression works best for photographs. Only RGB images are
+        supported. Note that compressing to JPEG costs a bit of CPU time, both
+        when logging and later when viewing them.
+
+        Parameters
+        ----------
+        jpeg_quality:
+            Higher quality = larger file size. A quality of 95 still saves a lot
+            of space, but is visually very similar.
+        """
+
+        from PIL import Image as PILImage
+
+        from .._image import ImageEncoded
+        from . import Image
+
+        self = cast(Image, self)
+
+        with catch_and_log_exceptions(context="Image compression"):
+            tensor_data_arrow = self.data.as_arrow_array()
+
+            if tensor_data_arrow[0].value["buffer"].type_code == self.JPEG_TYPE_ID:
+                _send_warning(
+                    "Image is already compressed as JPEG. Ignoring compression request.",
+                    1,
+                    recording=None,
+                )
+                return self
+
+            shape_dims = tensor_data_arrow[0].value["shape"].values.field(0).to_numpy()
+            non_empty_dims = find_non_empty_dim_indices(shape_dims)
+            filtered_shape = shape_dims[non_empty_dims]
+            if filtered_shape[-1] != 3:
+                raise ValueError("Only RGB images are supported for JPEG compression")
```

Comment:
The viewer supports decoding grayscale jpegs too; maybe we should add support for that here while we're at it. Grayscale images are very common from IR cameras, high-frame-rate cameras, etc